packages/flutter/lib/src/rendering/object.dart

```diff
@@ -2300,12 +2300,50 @@ abstract class RenderObject extends AbstractNode with DiagnosticableTreeMixin im
     }
   }
 
+  Set<SemanticsAction> _mutedSemanticsActions = <SemanticsAction>{};
+
+  /// Update the set of [SemanticsAction] that will be ignore when constructing
+  /// semantics action handlers.
+  ///
+  /// The muted [SemanticsAction] will apply to all the children in subtree.
+  void updateMutedSemanticsActions(Set<SemanticsAction> actions) {
```

Comment:
I wonder if RenderIgnorePointer could set "blockedActions" on the SemanticsConfiguration it returns from describeSemanticsConfiguration instead of adding this API here. The semantics compiler would than have to resolve this as it rebuilds the tree based on that changed configuration...

Reply:
Is this possible? We can do this if we calculate the subtree's blockedActions without any caching. If we want to use cache, we need a way to invalidate subtree's caches. I can't think of good way to do that if we didn't do it in renderobject level.

Reply:
It may make more sense to cache these things on the SemanticsNode itself. We already do this for other things, like the elevation imposed by a parent on its children or the parent paint clip: 
https://master-api.flutter.dev/flutter/semantics/SemanticsNode/elevationAdjustment.html
https://master-api.flutter.dev/flutter/semantics/SemanticsNode/parentPaintClipRect.html

It would probably be more idiomatic to have specify the blocked actions on the SemanticsConfiguration and cache things on the SemanticsNode.

Reply:
those case are setting semantics configuration for compilechildren to finalize the value. However, the blocked action I did here is blocking during semantics configuration setup. The biggest difference is that make node dirty will re compile its children SemanticsFragment but it does keep the old cache of semantics configuration. in the case of elevationAdjustment or parentPainterClipRect, it is ok because the subtree semantics configuration did not change. it just need to recompile. However, in my case it will break because the subtree need to reconfig semantics configuration. I think if we don't want to recursively mark subtree dirty, we can try to strip out blocked semanticsactions during compilechildren. I am not sure if there is any side effect tho.